homeassistant/components/mqtt_statestream/__init__.py

```diff
@@ -81,5 +101,21 @@ async def _state_publisher(entity_id, old_state, new_state):
                 encoded_val = json.dumps(val, cls=JSONEncoder)
                 await mqtt.async_publish(hass, mybase + key, encoded_val, 1, True)
 
-    async_track_state_change(hass, MATCH_ALL, _state_publisher)
+    if hass.state == CoreState.running:
+        ha_started.set()
+    else:
+
+        @callback
+        def _ha_started(_: Event) -> None:
```

Comment:
By waiting till HA is started, you're going to miss a ton of states. Now that seems to be your point, because you don't want to congest startup. However, that sounds like a restart is going to cause a ton of states to be missed now.

Note, for `mqtt_evenetstream`, we even make sure [it's set up prior to any other integration](https://github.com/home-assistant/core/blob/dev/homeassistant/bootstrap.py#L98-L99) to make sure we do not miss an event.

Reply:
Okay, I get your point. It seems for the issue that it is especially the shutting down that is causing issues, so I'll focus on that.

Reply:
Now the scope of operation is that we listen from the moment MQTT is available till HA is stopping.